datasource/src/main/scala/quasar/physical/blobstore/BlobstoreDatasource.scala

```diff
@@ -20,22 +20,28 @@ import slamdata.Predef._
 import quasar.api.datasource.DatasourceType
 import quasar.api.resource.{ResourceName, ResourcePath, ResourcePathType}
 import quasar.blobstore.Blobstore
-import quasar.connector.{MonadResourceErr, QueryResult, ResourceError}
+import quasar.connector.{MonadResourceErr, ParsableType, QueryResult, ResourceError}, ParsableType.JsonVariant
 import quasar.connector.datasource.LightweightDatasource
 import quasar.contrib.scalaz.MonadError_
 import quasar.contrib.std.errorNotImplemented
 
 import cats.Applicative
 import cats.effect.IO
-import fs2.Stream
+import cats.syntax.applicative._
+import fs2.{RaiseThrowable, Stream}
 
-class BlobstoreDatasource[F[_]: Applicative: MonadResourceErr](
+class BlobstoreDatasource[F[_]: Applicative: MonadResourceErr: RaiseThrowable](
   val kind: DatasourceType,
   blobstore: Blobstore[F])
   extends LightweightDatasource[F, Stream[F, ?], QueryResult[F]] {
 
-  override def evaluate(path: ResourcePath): F[QueryResult[F]] =
-    errorNotImplemented
+  val jvar = JsonVariant.LineDelimited
+
+  override def evaluate(path: ResourcePath): F[QueryResult[F]] = {
+    val bytes = blobstore.get(path)
+      .flatMap(buf => Stream.emits(buf.array()))
+    QueryResult.typed(ParsableType.json(jvar, false), bytes).pure[F]
```

Comment:
does it make sense to change `data` in
```
final case class Typed[F[_]](tpe: ParsableType, data: Stream[F, Byte]) extends QueryResult[F]
```
to `Stream[F, ByteBuffer]`?

Reply:
No. But what you do want to do is `Stream.chunk(Chunk.ByteBuffer(buf))` rather than turning the byte buffer into an array. The element type of `Stream` is *phantom*, in that those elements do not need to be efficiently directly accessible from the chunk type, assuming that the user isn't doing something like `map` or `filter` or some other element-level operation without first calling `chunks`.

Anyway, this in turn results in better performance since then fs2 is able to use different implementations of `Chunk` depending on exactly what operations you *do* perform. The canonical example here is operations which end up delegating to chunk append, which can be more efficiently handled by `ByteVector` than `ByteBuffer`. This also simplifies the types and avoids over-specializing.

Reply:
@djspiewak Oh, good to know, thanks! 

So do I understand it correctly that `renderJson` and `renderCsv` in `TableModule` are returning `StreamT[IO, CharBuffer]` and not `StreamT[IO, Char]` because this is a not an `fs2.Stream`? If they'd be returning an `fs2.Stream` then it would become a `Char` instead of a `CharBuffer`?

Reply:
In theory yes, though I don't think that there's a `Chunk.CharBuffer` type in fs2 (there probably should be). I think that more likely they would return `Stream[IO, CharBuffer]` which would then immediately be converted to a `Stream[IO, Byte]` (with bytebuffer chunks).